packages/compile-common/src/shims/FromHardhat.ts

```diff
@@ -0,0 +1,183 @@
+import type * as Hardhat from "hardhat/types";
+
+import type {
+  Compilation,
+  CompiledContract,
+  Bytecode,
+  LinkReference
+} from "../types";
+
+const supportedFormats = new Set(["hh-sol-build-info-1"]);
+
+export function buildInfoCompilation(
+  buildInfo: Hardhat.BuildInfo
+): Compilation {
+  const { _format } = buildInfo;
+
+  if (!supportedFormats.has(_format)) {
+    throw new Error(`Unsupported build info format: ${_format}`);
```

Comment:
<3 

It's great that you are validating this!

Reply:
Of course :D